npm/private/npm_import.bzl

```diff
@@ -450,17 +437,19 @@ def _download_and_extract_archive(rctx, package_json_only):
         canonical_id = download_url,
     )
 
-    mkdir_args = ["mkdir", "-p", _EXTRACT_TO_DIRNAME] if not repo_utils.is_windows(rctx) else ["cmd", "/c", "if not exist {extract_to_dirname} (mkdir {extract_to_dirname})".format(extract_to_dirname = _EXTRACT_TO_DIRNAME.replace("/", "\\"))]
+    is_windows = repo_utils.is_windows(rctx)
+
+    mkdir_args = ["mkdir", "-p", _EXTRACT_TO_DIRNAME] if not is_windows else ["cmd", "/c", "if not exist {extract_to_dirname} (mkdir {extract_to_dirname})".format(extract_to_dirname = _EXTRACT_TO_DIRNAME.replace("/", "\\"))]
     result = rctx.execute(mkdir_args)
     if result.return_code:
         msg = "Failed to create package directory. '{}' exited with {}: \nSTDOUT:\n{}\nSTDERR:\n{}".format(" ".join(mkdir_args), result.return_code, result.stdout, result.stderr)
         fail(msg)
 
     # npm packages are always published with one top-level directory inside the tarball, tho the name is not predictable
     # so we use tar here which takes a --strip-components N argument instead of rctx.download_and_extract
-    tar_args = ["tar", "-xf", _TARBALL_FILENAME, "--strip-components", str(1), "-C", _EXTRACT_TO_DIRNAME, "--no-same-owner", "--no-same-permissions"]
+    tar_args = ["tar", "-xf", _TARBALL_FILENAME, "--strip-components", "1", "-C", _EXTRACT_TO_DIRNAME, "--no-same-owner", "--no-same-permissions"]
 
-    if _is_gnu_tar(rctx):
+    if rctx.attr.is_gnu_tar == "True" or (rctx.attr.is_gnu_tar == "" and check_is_gnu_tar(rctx)):
```

Comment:
Maybe assign this to a var? 🤷

Reply:
Is this what you meant?
```
    is_gnu_tar = rctx.attr.is_gnu_tar
    if is_gnu_tar == "True" or (is_gnu_tar == "" and check_is_gnu_tar(rctx)):
```

The if condition needs to stay like this to short-circuit the evaluation

Reply:
Shouldn't it also be short-circuited when assigned to a var?

Reply:
I split out a var to make it a bit shorter. Splitting another one will just make it longer again I think? Check the current code

Reply:
I guess I'm just being picky/opinionated, but imo it just reads nicer when "compute `is_gnu_tar`" and `if is_gnu_tar` are two separate statements 🤷 

```
is_gnu_tar = ...
if is_gnu_tar:
  ...
```

Reply:
ok sure, i don't feel that strongly